src/geometrybasics_api.jl

```diff
@@ -1,62 +1,39 @@
+function GB.Mesh(f::Function,
+                 h::GB.Rect,
+                 samples::NTuple{3,T},
+                 method::AbstractMeshingAlgorithm;
+                 pointtype=GB.Point{3,Float32},
+                 facetype=GB.GLTriangleFace) where {T <: Integer}
+    vts, fcs = isosurface(f, method, pointtype, facetype; samples=samples,
+                          origin=pointtype(GB.origin(h)), widths=pointtype(GB.widths(h)))
 
-    vertex_eltype = promote_type(FT, typeof(method.iso), typeof(method.eps))
-    VertType, FaceType = _determine_types_gb(pointtype, facetype, vertex_eltype,
-                                          GB.default_face_length(method))
-    h = df.bounds
-    vts, fcs = isosurface(df.data, method, VertType, FaceType, origin=VertType(GB.origin(h)), widths=VertType(GB.widths(h)))
-    return GB.Mesh(vts, fcs)
+    return GB.Mesh(GB.meta(vts), fcs)
 end
 
-function GB.mesh(f::Function, h::GB.Rect, samples::NTuple{3,T}, method::AbstractMeshingAlgorithm;
-              pointtype=nothing, facetype=nothing) where {T <: Integer}
-    vertex_eltype = promote_type(T, typeof(method.iso), typeof(method.eps))
-    VertType, FaceType = _determine_types_gb(pointtype, facetype, vertex_eltype,
-                                          default_face_length(method))
-    vts, fcs = isosurface(f, method, VertType, FaceType; samples=samples,
-                          origin=VertType(GB.origin(h)), widths=VertType(GB.widths(h)))
-    return GB.Mesh(vts, fcs)
-end
+function GB.Mesh(f::Function, h::GB.Rect, method::AbstractMeshingAlgorithm;
+              samples::NTuple{3,T}=_DEFAULT_SAMPLES,
+              pointtype=GB.Point{3,Float32},
+              facetype=GB.GLTriangleFace) where {T <: Integer}
 
-function GB.mesh(f::Function, h::GB.Rect, method::AbstractMeshingAlgorithm;
-              samples::NTuple{3,T}=_DEFAULT_SAMPLES, pointtype=nothing,
-              facetype=nothing) where {T <: Integer}
-    return mesh(f, h, samples, method; pointtype=pointtype, facetype=facetype)
+    return GB.Mesh(f, h, samples, method; pointtype=pointtype, facetype=facetype)
 end
 
-function GB.mesh(volume::AbstractArray{T, 3}, method::AbstractMeshingAlgorithm; pointtype=nothing, facetype=nothing, kwargs...) where {T}
-    vertex_eltype = promote_type(T, typeof(method.iso), typeof(method.eps))
-    VertType, FaceType = _determine_types_gb(pointtype, facetype, vertex_eltype,
-                                          default_face_length(method))
-    vts, fcs = isosurface(volume, method, VertType, FaceType; kwargs...)
-    return GB.Mesh(vts, fcs)
+function GB.Mesh(volume::AbstractArray{T, 3},
+                 method::AbstractMeshingAlgorithm;
+                 pointtype=GB.Point{3,Float32},
+                 facetype=GB.GLTriangleFace,
+                 kwargs...) where {T}
+
+    vts, fcs = isosurface(volume, method, pointtype, facetype; kwargs...)
+    return GB.Mesh(GB.meta(vts), fcs)
```

Comment:
Why do you call meta on the points?

Reply:
It is not needed?

Reply:
no, shouldn't... You just need that, to add additional meta information